overlord/servicestate/quotas.go

```diff
@@ -29,6 +29,16 @@ import (
 
 var ErrQuotaNotFound = errors.New("quota not found")
 
+// AllQuotas returns all currently tracked quota groups in the state. They are
+// validated for consistency using ResolveCrossReferences before being returned.
+// API request should only succeed if the cgroups are enabled in the system
+func AllQuotasForDaemon(st *state.State) (map[string]*quota.Group, error) {
```

Comment:
It's not clear what is the difference between this and `AllQuotas`: judging by the function name, it would appear that this one returns all quotas for the snapd process, whereas `AllQuotas()` returns quotas for all installed snaps. But looking at the code this is not true, and the only difference is that this returns an error if quotas are not enabled.

Are you sure that we need to different functions? Would it break many things if AllQuotas returned an error? Who is calling it?

Reply:
AllQuotas serve two purposes; one is to retrieve quotas for requests coming through the API, the second is giving that information to the managers in most snap operations. In case of an error in AllQuotas snap removal and refresh both fail so we decided to decouple the two usages for ease of development as we can also keep a single function and differentiate the source to check for permissions.